### PR TITLE
Fixed TTL settings. Updated docs.

### PR DESCRIPTION
**Pull request by arb**
- Added some randomness to the ttl in the cache headers. 
- Documented this change.
- Updated tests.
- Added check in "onPreResponse" extension point that was really missing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ server.register({
...
+When the first request comes in to "/", the `hydrate` method is called. We are getting the Google home page and after 1000 milliseconds, we are calling back with the result. If you make a second request to "/", you should notice the delay isn't there and the response is almost instant. This is because the original response has been cached and sent back to the client. If you are testing with a browser, you should notice that the cache header decrements on each request. tacky will set the client cache header ("cache-control:max-age=3566, must-revalidate") based on the ttl options. The cache header ttl will be randomized so that the server isn't slammed by multiple requests at the same time. The goal is to stagger the cache header expiration across different clients.
```
Comment (cjihrig) [inline README.md]: `almost instant`. Should this be `almost instantaneous`?

File: test/index.js [hunk 2]
```diff
@@ -14,6 +14,16 @@ var before = lab.before;
 var after = lab.after;
 var expect = Code.expect;
 
+var internals = {
+    headerRegex: /max-age=\d+,\s+must-revalidate(,\s+private)?/
+};
+
+internals.getTtl = function (header) {
+    var age = header.match(/(\d+)/)[0];
```
Comment (cjihrig) [inline test/index.js]: Maybe you don't need to be too bulletproof since this is test code, but `match()` can return `null`.

File: test/index.js [hunk 3]
```diff
@@ -102,9 +115,10 @@ describe('tacky', function () {
                         url: '/'
                     }, function (res) {
 
-                        expect(res.headers['cache-control']).to.equal('max-age=100, must-revalidate');
+                        expect(res.headers['cache-control']).to.exist();
+                        expect(internals.headerRegex.test(res.headers['cache-control'])).to.be.true();
```
Comment (cjihrig) [inline test/index.js]: You can use `expect(value).to.match(regex)` here and in other places.

Comment (arb) [inline test/index.js]: Good call! I forgot about that one.
